### PR TITLE
Implement GUI testing using TestFX

### LINKED ISSUE #54: Implement GUI tests
Testing GUI is annoying, let's automate it with the testFX library! This will make it easier to test GUI in the future as well as increase our codecov score.

### LINKED ISSUE #60: Update archiveName property in build.gradle file
Screenshot 2021-03-03 at 2 49 23 PM

archiveName property has been deprecated in favour of archiveFileName.

### PR DESCRIPTION
**Pull request by samuelfangjw**
closes #54, closes #60

### CODE REVIEW

**Comment by samuelfangjw** [comment]
Need to merge #51 before merging this PR, also need to request to use TestFX library on the forum.

File: src/test/java/guitests/guihandles/PersonListPanelHandle.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+    private Optional<Person> lastRememberedSelectedPersonCard;
+
+    public PersonListPanelHandle(ListView<Person> personListPanelNode) {
+        super(personListPanelNode);
+    }
+
+    /**
+     * Returns a handle to the selected {@code PersonCardHandle}.
+     * A maximum of 1 item can be selected at any time.
```
Comment (vevek) [inline src/test/java/guitests/guihandles/PersonListPanelHandle.java:30]: Add extra line here

File: src/test/java/guitests/guihandles/PersonListPanelHandle.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    /**
+     * Selects the {@code PersonCard} at {@code index} in the list.
+     */
+    public void select(int index) {
+        getRootNode().getSelectionModel().select(index);
+    }
+
+    /**
+     * Returns the person card handle of a person associated with the {@code index} in the list.
+     * @throws IllegalStateException if the selected card is currently not in the scene graph.
```
Comment (vevek) [inline src/test/java/guitests/guihandles/PersonListPanelHandle.java:107]: Extra line here.

File: src/test/java/guitests/guihandles/StatusBarFooterHandle.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+public class StatusBarFooterHandle extends NodeHandle<Node> {
+    public static final String STATUS_BAR_PLACEHOLDER = "#statusbarPlaceholder";
+
+    private static final String SAVE_LOCATION_STATUS_ID = "#saveLocationStatus";
+
+    private final Labeled saveLocationNode;
+
+    /**
+     * Constructs a {@code StatusBarFooterHandle} handler object.
+     * @param statusBarFooterNode Node of the {@code StatusBarFooter}.
```
Comment (vevek) [inline src/test/java/guitests/guihandles/StatusBarFooterHandle.java:22]: Add extra line here.

**Review by vevek — DISMISSED** [review]
LGTM!

**Review by vevek — APPROVED** [review]
LGTM!

**Pull request merged**